modules/pools/components/PoolListItem.tsx

```diff
@@ -15,6 +15,10 @@ interface Props extends BoxProps {
 }
 
 export function PoolListItem({ pool, userBalance, showUserBalance, ...rest }: Props) {
+    const tokenData = pool.allTokens
+        .filter((token) => !token.isNested && !token.isPhantomBpt)
+        .map((token) => ({ address: token.address, weight: token.weight } as TokenData));
```

Comment:
Forcing a type here is incorrect, the problem is your data does not align to your model, forcing the type is just hiding the problem